mozilla/blurts-server
# Don't count manual resolves as unresolved

# References: 
Jira: MNTOR-2268
Figma: N/A




# Description

This PR does _three_ things. The most important thing is that manual resolutions are now subtracted from the count of the "Action needed" dashboard: 1a58a4e8daac0c27bbae6c31e3329f701ac63fa4.

That is the part that fixes the issue. Most of the code changes, however, are renames (b0caf54218fe1545d09b638323b226dd8de00d41) and a small bit of refactoring (d1c0498b1e8f8805edd52d60cb843d8d8485f79c). I had to do those to be able to understand what was going on and how to fix it. The main thing with big impact is that I renamed `exposure` in `dashboard.ts` to `dataPoint`, because we already used "exposure" to mean "broker scan result or data breach" elsewhere in the code, and I kept getting confused. **I would recommend filtering the review by commits** to ease review.

And then finally, there's **clearing the router cache** (c2a5769ac68c1d96bb026ef480cb0a4525963482). This solves the issue where, going back to the dashboard, the numbers would not actually be updated. ([See docs](https://nextjs.org/docs/app/building-your-application/caching#invalidation-1).) We will probably need to use that elsewhere as well?

# How to test

Manually resolve a data broker scan result, then visit the dashboard and verify that it is updated.

# Checklist (Definition of Done)
- [x] I've added a unit test to test for potential regressions of this bug. - Sort of - I fixed an existing incorrect unit test.
